pkg/reconciler/openshift/openshift.go

```diff
@@ -18,7 +18,7 @@ func init() {
 func reconcilerHook(cr *argoprojv1alpha1.ArgoCD, v interface{}) error {
 	switch o := v.(type) {
 	case *rbacv1.ClusterRole:
-		if o.ObjectMeta.Name == cr.ObjectMeta.Name+"-argocd-application-controller" {
+		if o.ObjectMeta.Name == cr.ObjectMeta.Name + cr.ObjectMeta.Namespace + "-argocd-application-controller" {
```

Comment:
Please call the "generateClusterResourceName"

Reply:
@wtam2018, is it OK if we make this change as part of refactoring? We could probably create exported functions `GenerateResourceName` and `GenerateClusterResourceName` in the utils package and use them across the code rather than changing it just here?

Reply:
Done !!!

Reply:
Also, updated the function name from `GenerateClusterResourceName` to `GenerateUniqueResourceName`, as that makes more sense.